extensions/amp-redbull-player/0.1/amp-redbull-player.js

```diff
@@ -0,0 +1,298 @@
+import getPlayerParams from './utils/getPlayerParams';
+
+/** @private @const */
+const TAG = 'amp-redbull-player';
+
+/** @private @const */
+const ANALYTICS_EVENT_TYPE_PREFIX = 'video-custom-';
+
+/** @private @const */
+const SANDBOX = [
+  SandboxOptions.ALLOW_SCRIPTS,
+  SandboxOptions.ALLOW_SAME_ORIGIN,
+  SandboxOptions.ALLOW_POPUPS,
+  SandboxOptions.ALLOW_POPUPS_TO_ESCAPE_SANDBOX,
+  SandboxOptions.ALLOW_TOP_NAVIGATION_BY_USER_ACTIVATION,
+];
+
+/**
+ * @return {!RegExp}
+ * @private
+ */
+const getAnalyticsEventTypePrefixRegex = once(
+  () => new RegExp(`^${ANALYTICS_EVENT_TYPE_PREFIX}`)
+);
+
+/** @implements {../../../src/video-interface.VideoInterface} */
+class AmpRedBullPlayer extends AMP.BaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    const el = addAttributesToElement(element);
+    super(el);
+
+    /** @private {?Element} */
+    this.iframe_ = null;
+
+    /** @private {!UnlistenDef|null} */
+    this.unlistenFrame_ = null;
+
+    /** @private {string} */
+    this.tagId_ = null;
+
+    /** @private {string} */
+    this.locale_ = null;
+
+    /**
+     * @param {!Event} e
+     * @return {*} TODO(#23582): Specify return type
+     * @private
+     */
+    this.boundOnMessage_ = e => this.onMessage_(e);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  buildCallback() {
+    const {element} = this;
+    /*
+     * Required Paramaters: data-video-id, data-skin
+     * Break creation if required parameters are missing!
+     */
+    userAssert(
+      element.getAttribute('data-videoid'),
+      'The data-videoid attribute is required for %s',
+      element
+    );
+
+    userAssert(
+      element.getAttribute('data-skin'),
+      'The data-skin attribute is required for %s',
+      element
+    );
+    this.tagId_ = element.getAttribute('id') || '';
+    this.locale_ = element.getAttribute('data-locale') || 'global';
+
+    installVideoManagerForDoc(element);
+  }
+
+  /** @override */
+  layoutCallback() {
+    const {element} = this;
+    const origin = 'https://player.redbull.com/amp/amp-iframe.html?';
+    // const origin = 'http://localhost:9000/amp-iframe.html?';
+    const params = `${getPlayerParams(element)}&ampTagId=${
+      this.tagId_
+    }&locale=${this.locale_}`;
+    const src = `${origin}${params}`;
+
+    const name = JSON.stringify(this.getMetadata_());
+    this.iframe_ = disableScrollingOnIframe(
+      createFrameFor(this, src, name, SANDBOX)
+    );
+
+    this.unlistenFrame_ = listen(this.win, 'message', this.boundOnMessage_);
+    const loaded = this.loadPromise(this.iframe_).then(() => {
+      this.onReady_();
+    });
+    return loaded;
+  }
+
+  /** @private */
+  onReady_() {
+    const {element} = this;
+    Services.videoManagerForDoc(element).register(this);
+
+    this.sendCommand_({msg: 'amp-loaded'});
+    element.dispatchCustomEvent(VideoEvents.LOAD);
```

Comment:
nit: just use `this.element` directly since there's only one reference. Usually we destructure to reduce binary size on multiple references, but this is just one.